gazebo/create_dockerfiles.py

```diff
@@ -16,6 +19,35 @@
 from ros_buildfarm.docker_common import OrderedLoad
 
 
+def checkVersion(data):
+    """Checks current package_version from online package index"""
+
+    # Determine URL
+    urlTemplate = string.Template(
+        """
+        http://packages.osrfoundation.org/gazebo/$os_name/dists/$os_code_name/main/binary-$arch/Packages
+        """)
+    url = urlTemplate.substitute(data)
+
+    # Download package index
+    req = urllib.request.Request(url)
+    with urllib.request.urlopen(req) as response:
+        package_index = response.read().decode('utf-8')
+
+    # Determine seach pattern
+    patternTemplate = string.Template(r'(\bPackage: gazebo$version\n)(.*\n)')
+    pattern_raw = patternTemplate.substitute(data)
+    pattern = re.compile(pattern_raw)
+
+    # Parse for version_number
+    matchs = re.search(pattern, package_index)
+    version_line = matchs.groups(0)[1] # Grab the second line of the first match
+    version_number = re.search(r'\d{1}[.]\d{1}[.]\d{1}', version_line).group(0) # extract version_number
```

Comment:
It's true that we keep a weird version name (`$version + ~$distro`) but we probably need to cover more use cases than just the standard `d.d.d`. I would suggest to get all the field and exclude just (and only) the final ~ and what comes next until the end. I.e: `8.0.0~beta4~yakkety` to be `8.0.0~beta4`.

You can try it replacing the stable URL above by `$os_name-prerelease`